### PR TITLE
[SG-702] Tapping Push Notification does not open the account the request is for

### PR DESCRIPTION
**Pull request by andrebispo5**
## Type of change
- [X] Bug fix
- [ ] New feature development
- [ ] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other

## Objective

Tapping Push Notification does not take you to account the request is for.

## Code changes

Added new method in `src/App/Services/PushNotificationListenerService.cs` to handle notification tap logic.

## Before you submit
- Please check for formatting errors (`dotnet format --verify-no-changes`) (required)
- Please add **unit tests** where it makes sense to do so (encouraged but not required)
- If this change requires a **documentation update** - notify the documentation team
- If this change has particular **deployment requirements** - notify the DevOps team

### CODE REVIEW

File: src/App/Services/PushNotificationListenerService.cs [hunk 1]
```diff
@@ -213,6 +218,29 @@ public void OnError(string message, string deviceType)
...
+                    if (notificationUserId != null)
+                    {
+                        await _stateService.SetActiveUserAsync(notificationUserId);
+                        _messagingService.Send(AccountsManagerMessageCommands.SWITCHED_ACCOUNT);
+                    }
+                }
+            }
+            catch (Exception ex)
+            {
+                LoggerHelper.LogEvenIfCantBeResolved(ex);
```
Comment (fedemkr) [inline src/App/Services/PushNotificationListenerService.cs:240]: If possible inject the `ILogger` and use it instead of the helper

**Review by fedemkr — CHANGES_REQUESTED** [review]

File: src/Android/MainActivity.cs [hunk 2]
```diff
@@ -145,6 +147,11 @@ protected override void OnResume()
             AndroidHelpers.SetPreconfiguredRestrictionSettingsAsync(this)
                 .GetAwaiter()
                 .GetResult();
+
+            if (Intent?.GetStringExtra(Constants.PasswordlessNotificationType) is string notificationDataJson)
+            {
+                _pushNotificationListenerService.OnNotificationTapped(Constants.PasswordlessNotificationType, notificationDataJson).FireAndForget();
+            }
```
[the remark this answers is not in the document]
Comment (andrebispo5) [reply src/Android/MainActivity.cs:154]: Yes, if the app is in the background and the user clicks the notification it won’t pass on `OnCreate` or `NewIntent`

File: src/App/Abstractions/IPushNotificationService.cs [hunk 3]
```diff
@@ -10,7 +10,7 @@ public interface IPushNotificationService
         Task<string> GetTokenAsync();
         Task RegisterAsync();
         Task UnregisterAsync();
-        void SendLocalNotification(string title, string message, string notificationId);
+        void SendLocalNotification(string title, string message, string notificationId, string notificationType, Dictionary<string, string> notificationData = null, int notificationTimeoutMinutes = 0);
```
[the remark this answers is not in the document]
Comment (andrebispo5) [reply src/App/Abstractions/IPushNotificationService.cs]: Did a big rework for this one.

File: src/App/Models/NotificationData.cs [hunk 4]
```diff
@@ -3,21 +3,20 @@ namespace Bit.App.Models
 {
     public abstract class BaseNotificationData
     {
-        public abstract string Type { get; }
+        public abstract string NotificationType { get; }
 
+        public string NotificationId { get; set; }
```
Comment (fedemkr) [inline src/App/Models/NotificationData.cs:8]: IMHO there is no need to add the prefix "Notification" to the properties given that the class is called notification data so adding the prefix to anything in there is redundant. Or was is confusing on some context?
Also, for now we can leave it as is but in the future if there are notifications that don't require additional data we should extract the `Id` out of this class (which is the only thing required).

File: src/App/Abstractions/IPushNotificationListenerService.cs [hunk 5]
```diff
@@ -9,7 +10,7 @@ public interface IPushNotificationListenerService
         Task OnRegisteredAsync(string token, string device);
         void OnUnregistered(string device);
         void OnError(string message, string device);
-        Task OnNotificationTapped(string type, string data);
+        Task OnNotificationTapped(string type, BaseNotificationData data);
```
Comment (fedemkr) [inline src/App/Abstractions/IPushNotificationListenerService.cs:13]: if I'm not mistaken, `type` is no longer needed

**Review by fedemkr — CHANGES_REQUESTED** [review]

**Review by fedemkr — APPROVED** [review]

**Pull request merged**